### PR TITLE
RELATED: RAIL-4068 Adjust tsdoc for better API reference formatting

### PR DESCRIPTION
**Pull request by plavemar**
JIRA: RAIL-4068

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-backend-spi/src/workspace/styling/theme.ts [hunk 1]
```diff
@@ -326,82 +778,17 @@ export interface ITheme {
     /**
      * Button customizable UI properties
      */
-    button?: {
```
Comment (kandl) [inline libs/sdk-backend-spi/src/workspace/styling/theme.ts:329]: Should it be named rather "IThemeButton" to make it consistent with the previous parts?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7284/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9136/ : SUCCESS in 16m 41s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7285/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9137/ : SUCCESS in 16m 45s

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7312/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9168/ : SUCCESS in 17m 58s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7315/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9173/ : SUCCESS in 18m 35s

**Review by ivanmjartan — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2349

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2115/ : SUCCESS in 1m 03s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2399/ : SUCCESS in 35s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1218/ : SUCCESS in 20m 03s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9174/ : FAILURE in 25m 23s

File: libs/sdk-backend-spi/src/workspace/execution/results.ts [hunk 2]
```diff
@@ -1,27 +1,86 @@
...
-    };
+    measureGroupHeader: IMeasureDescriptorObject;
+}
+
+/**
+ * Measure descriptor object.
+ *
+ * @public
+ */
+export interface IMeasureHeaderItem {
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/execution/results.ts:42]: Should be `IMeasureDescriptorItem`, we are trying to get rid of the overused header stuff.

File: libs/sdk-backend-spi/src/workspace/execution/results.ts [hunk 3]
```diff
@@ -77,85 +109,104 @@ export interface IMeasureDescriptor {
...
-    };
+    totalHeaderItem: ITotalDescriptorItem;
+}
+
+/**
+ * Describes attributes to which the display form belongs.
+ *
+ * @public
+ */
+export interface IAttributeHeaderForm {
```
Comment (no23reason) [inline libs/sdk-backend-spi/src/workspace/execution/results.ts:120]: I would use `IAttributeHeaderFormOf`: this is not some form, but rather specification of which attribute this header display form is of. So maybe even `IAttributeHeaderAttribute`?

File: libs/sdk-ui-charts/src/charts/areaChart/AreaChart.tsx [hunk 4]
```diff
@@ -200,16 +203,19 @@ export interface IAreaChartProps extends IBucketChartProps, IAreaChartBucketProp
 const WrappedAreaChart = withChart(areaChartDefinition)(CoreAreaChart);
 
 /**
+ * Area chart shows data as an area under a line intersecting dots.
+ *
+ * @remarks
  * [AreaChart](http://sdk.gooddata.com/gooddata-ui/docs/area_chart_component.html)
```
Comment (no23reason) [inline libs/sdk-ui-charts/src/charts/areaChart/AreaChart.tsx:209]: Should probably be a `@link` right? Similarly in other chart types.

File: libs/sdk-ui/src/execution/withExecutionLoading.tsx [hunk 5]
```diff
@@ -104,8 +104,11 @@ export interface IWithLoadingEvents<TProps> {
 }
 
 /**
- * Configuration for the withLoading HOC. All configuration parameters can be either actual parameter values
- * or functions to obtain them from the wrapped component props.
+ * Configuration for the withLoading HOC.
```
Comment (no23reason) [inline libs/sdk-ui/src/execution/withExecutionLoading.tsx:107]: Please `@link` to the HOC.

File: libs/sdk-backend-spi/src/workspace/styling/theme.ts (hunk 1 above)
Comment (plavemar) [reply libs/sdk-backend-spi/src/workspace/styling/theme.ts:329]: All are renamed as recommended.

File: libs/sdk-ui-pivot/src/columnWidths.ts [hunk 6]
```diff
@@ -240,9 +282,9 @@ export function newMeasureColumnLocator(measureOrId: IMeasure | string): IMeasur
 /**
  * Creates width item that will set width of a column which contains values of a row attribute.
  *
- * @param attributeOrId - attribute specified by value or by localId reference
- * @param width - width in pixels
- * @param allowGrowToFit - indicates whether the column is allowed to grow if the table's growToFit is enabled
+ * @param attributeOrId - Attribute specified by value or by localId reference
+ * @param width - Width in pixels
+ * @param allowGrowToFit - Optionally indicates whether the column is allowed to grow if the table's growToFit is enabled
```
[the remark this answers is not in the document]
Comment (plavemar) [reply libs/sdk-ui-pivot/src/columnWidths.ts:287]: In param, we can keep these "Optionally" as there is no optionality generated in the documentation. In params, on the other hand, we should keep this info in the `@param` tags

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7327/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9183/ : SUCCESS in 17m 20s

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2349

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2118/ : SUCCESS in 1m 10s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2402/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1221/ : SUCCESS in 17m 41s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9187/ : SUCCESS in 21m 30s

**Pull request merged**